### PR TITLE
Purge snapshots

### PR DESCRIPTION
**Pull request by ccomb**
Implemented the purge command and schedule, and improved the overall documentation

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -59,24 +62,223 @@ Or directly by pulling a `prebaked image <https://hub.docker.com/r/anybox/butter
...
+    $ buttervolume send <host> <snapshot>
+
+``<snapshot>`` is the name of the snapshot, not the full path. It is expected
+to live in ``/var/lib/docker/snapshots``.
+
+
+``<host>`` is the hostname or IP address of the remote host. The snapshot is
+currently sent using BTRFS send/receive through ssh. This requires that ssh
+keys be present and already authorized on the target host, and that the
+``StrictHostKeyChecking no`` option be enabled in ``~/.ssh/config``.
```
Comment (petrus-v) [inline README.rst]: we could add that the snapshot are sent to ``/var/lib/docker/snapshots`` to the host, insn't it?

File: buttervolume/plugin.py [hunk 2]
```diff
@@ -271,3 +271,63 @@ def snapshot_restore():
...
+    volume_name = params['Name']
+    dryrun = params.get('Dryrun', False)
+    units = {'m': 1, 'h': 60, 'd': 60*24, 'w': 60*24*7, 'y': 60*24*365}
+    try:
+        pattern = sorted(int(i[:-1])*units[i[-1]]
+                         for i in params['Pattern'].split(':'))
+        assert(len(pattern) >= 2)
+        max_age = pattern[-1]
+    except:
+        return {'Err': 'Invalid purge pattern'}
```
Comment (petrus-v) [inline buttervolume/plugin.py]: should we use json.dump?

File: buttervolume/plugin.py [hunk 3]
```diff
@@ -271,3 +271,63 @@ def snapshot_restore():
...
+    # snapshots related to the volume, more recents first
+    snapshots = sorted([s for s in os.listdir(SNAPSHOTS_PATH)
```
Comment (petrus-v) [inline buttervolume/plugin.py]: not sure with python 3.5 I through sorted do not return a new list but change the list in place

File: buttervolume/plugin.py [hunk 4]
```diff
@@ -271,3 +271,63 @@ def snapshot_restore():
...
+                        if s.startswith(volume_name + '@')], reverse=True)
+    now = datetime.now()
+    # Age of the snapshots in minutes.
+    # Example : [30, 70, 90, 150, 210, ..., 4000]
+    snapshots_age = [int((now - datetime.strptime(
+                        s.split('@')[1],
+                        "%Y-%m-%dT%H:%M:%S.%f")).total_seconds())//60
+                     for s in snapshots]
+    if not snapshots:
+        return {'Err': ''}
```
Comment (petrus-v) [inline buttervolume/plugin.py]: json.dumps?

File: test.py [hunk 5]
```diff
@@ -381,6 +381,75 @@ def test_restore(self):
...
+                             json.dumps({'Name': name, 'Pattern': '2h:2h'}))
+        path = join(VOLUMES_PATH, name)
+        self.app.post('/VolumeDriver.Create', json.dumps({'Name': name}))
+        with open(join(path, 'foobar'), 'w') as f:
+            f.write('original foobar')
+
+        def cleanup_snapshots():
+            for s in os.listdir(SNAPSHOTS_PATH):
+                if s.startswith(PREFIX_TEST_VOLUME):
+                    btrfs.Subvolume(join(SNAPSHOTS_PATH, s)).delete()
```
Comment (petrus-v) [inline test.py]: as far I remember, I've a bit change that behaviour, you could use btrfs.Subvolume(join(SNAPSHOTS_PATH, PREFIX_TEST_VOLUME) + '*').delete()

File: buttervolume/plugin.py (hunk 2 above)
Comment (ccomb) [inline buttervolume/plugin.py]: yes, good catch

File: README.rst (hunk 1 above)
Comment (ccomb) [inline README.rst]: done

File: buttervolume/plugin.py (hunk 3 above)
Comment (ccomb) [inline buttervolume/plugin.py]: possible yes, anyway I removed the [ ]

File: buttervolume/plugin.py (hunk 4 above)
Comment (ccomb) [inline buttervolume/plugin.py]: +1

File: test.py (hunk 5 above)
Comment (ccomb) [inline test.py]: ok though this is not the cleanup of the setUp

File: buttervolume/plugin.py (hunk 3 above)
Comment (petrus-v) [inline buttervolume/plugin.py]: My bad I think I've mixed with [].sort()

**Pull request merged**